concent_api/utils/logging.py

```diff
@@ -337,3 +382,59 @@ def _get_message_type(
     message: Message
 ) -> str:
     return type(message).__name__ if isinstance(message, Message) else '-not available- '
+
+
+def is_redundant_callable_or_golem_messages_field(golem_message, field_name):
+    return False if isinstance(getattr(golem_message, field_name), Message) else callable(getattr(golem_message, field_name))
+
+
+def get_json_from_message_without_redundant_fields_for_logging(
+    golem_message: Message,
+) -> JsonResponse:
+
+    dictionary_to_serialize = serialize_message_to_dictionary(golem_message)
+
+    for field in dir(golem_message):
+        if isinstance(getattr(golem_message, field), TaskToCompute):
+            task_to_compute_dictionary = serialize_message_to_dictionary(getattr(golem_message, field))
+            dictionary_to_serialize.update({field: task_to_compute_dictionary})
+
+    return json.dumps(dictionary_to_serialize, indent=4)
+
+
+def serialize_message_to_dictionary(
+    golem_message: Message,
+)->dict:
+
+    fields_to_serialize = [f for f in dir(golem_message) if not f.startswith('_') and not f.isupper() and not is_redundant_callable_or_golem_messages_field(golem_message, f)]
```

Comment:
Please break it into separate lines.

Reply:
done. All conditions all now in one method `is_redundant_field`

Reply:
Just a suggestion - if you ever need to split a list comprehension, here's a readable way to do it:

``` python
fields_to_serialize = [
    field_name
    for field_name in dir(golem_message)
    if (
        not field_name.startswith('_') and
        not field_name.isupper() and
        not is_redundant_callable_or_golem_messages_field(golem_message, field_name)
    )
]
```

But yeah, creating a helper function for the conditions is a better idea.